src/notebooks/Python Examples/GSC23 MGnify Workshop.ipynb

```diff
@@ -0,0 +1,1529 @@
+    "\n",
+    "The next most common species occurs in the catalogues 3 times and there are at least 9 species like that as we can see from the table (there are likely more but we only printed the first 10 lines). Since our catalogues are dereplicated at the species level, each species can only be found in a biome-specific catalogue once. In total we currently have 9 catalogues so we wouldn't expect the maximum number of occurences for any species to exceed 9."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a4bf1bc-afb5-4c19-81a2-a3d51130edac",
+   "metadata": {},
+   "source": [
+    "##### Which catalogues contain the novel Prevotella species?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2664b705-3904-4c40-9ca0-701d12892398",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Save the rows containing novel Prevotella species into a new dataframe\n",
+    "novel_prevotella_df = resources_df[resources_df['attributes.taxon-lineage'] == 'd__Bacteria;p__Bacteroidota;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Prevotella;s__']\n",
+    "# Plot the catalogues\n",
+    "novel_prevotella_df.groupby('relationships.catalogue.data.id').size().plot(kind='pie', autopct='%1.0f%%')\n",
+    "plt.title('Catalogues where the novel Prevotella species are found');\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "93ad9b03-9cd6-411a-986a-625cb5f9afea",
+   "metadata": {},
+   "source": [
+    "##### Q: Which catalogue has a higher number of novel Prevotella species?  \n",
+    "\n",
+    "##### Q: Which catalogue only contains known Prevotella species?\n",
+    "Hint: compare this pie chart to the previous one - which catalogue disappeared?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eacde0da-c37f-4681-889d-f50b6b5112c1",
+   "metadata": {},
+   "source": [
+    "#### Task 3: Create a graph showing most common phyla in MGnify Genomes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cbae69d4-594e-414d-be2a-8a5383d091a7",
+   "metadata": {},
+   "source": [
+    "For this exercise we will be working with pre-fetched data for all genomes. This is because fetching it takes a while given the number of genomes but if you were fetching it yourself, the code for that is below (expand by clicking on the three dots but please don't execute the code)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d576840c-5535-4958-8031-5952c66532b8",
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "endpoint = \"genomes\"\n",
+    "with APISession(\"https://www.ebi.ac.uk/metagenomics/api/v1\") as mgnify:\n",
+    "    resources_all = map(lambda r: r.json, mgnify.iterate(endpoint))\n",
+    "resources_all_df = pd.json_normalize(resources_all)\n",
+    "resources_all_df.to_parquet('all_genome_resources_Aug2023.parquet')"
+   ]
+  },
```

Comment:
Maybe best to comment this out, or put it as code-in-markdown? I was a naughty user and just absent-mindedly clicked run!